### PR TITLE
DEV-1582: Adopt ML backend for cloud run

### PR DESCRIPTION
**Pull request by farioas**

### CODE REVIEW

**Comment by niklub** [comment]
Add Readme with required prerequisites, parameters and usage example

File: label_studio_ml/server.py [hunk 1]
```diff
@@ -111,13 +125,42 @@ def start_server(args, subprocess_params):
...
+    auth_token = subprocess.check_output(' '.join(["gcloud", "auth", "print-identity-token"]), shell=True)
+    if not auth_token:
+        raise PermissionError("You are not authentificated in gcloud! Please run gcloud auth login.")
+    # configurate project
+    subprocess.check_output(' '.join(["gcloud", "config", "set", "project", project_id]), shell=True)
+    # deploy service
+    subprocess.check_output(' '.join(["gcloud", "run", "deploy", service_name, "--source", output_dir, "--region", region]), input=b"y", shell=True)
+
+
+def special_match(strg, search=re.compile(r'[^a-z-]').search):
```
Comment (niklub) [inline label_studio_ml/server.py:151]: this one doesn't work for all types of projects, e.g. `aqueous-cortex-307813`

Comment (niklub) [reply label_studio_ml/server.py:151]: also please do not include object creation in default initializers

**Review by niklub — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -129,3 +129,24 @@ def fit(self, completions, workdir=None, **kwargs):
...
+```
+4. Activate your Cloud Build API
+5. Add GCP_PROJECT with your GCP project name to your ENV variables 
+6. (Optional) Add GCP_REGION with your default region to your ENV variables 
+
+To start deployment:
+1. Create your own ML backend
+2. Start deployment to GCP:
+```bash
+label-studio-ml deploy_gcp {project_name} --from={model_script}
```
Comment (niklub) [inline README.md:150]: could we support:
```
label-studio-ml deploy gcp {ml-backend-name} --gcp-project {project_name} --from={model_script}
```

**Review by niklub — CHANGES_REQUESTED** [review]

File: label_studio_ml/examples/simple_text_classifier/requirements.txt [hunk 3]
```diff
@@ -1 +1,7 @@
-scikit-learn==0.24.1
\ No newline at end of file
+# required
+gunicorn==20.1.0
+label-studio-ml==1.0.4
+rq==1.10.1
+
+# optional
```
Comment (niklub) [inline label_studio_ml/examples/simple_text_classifier/requirements.txt:6]: why it's optional?

File: label_studio_ml/server.py [hunk 4]
```diff
@@ -111,13 +131,48 @@ def start_server(args, subprocess_params):
     os.system('python ' + wsgi + ' ' + ' '.join(subprocess_params))
 
 
+def deploy_to_gcp(args):
+    # create project with
+    create_dir(args)
+    # prepare params for gcloud: dir with script, project id, region and service name
+    output_dir = os.path.join(args.root_dir, args.project_name)
+    project_id = os.environ.get("GCP_PROJECT")
```
Comment (niklub) [inline label_studio_ml/server.py:139]: let's make this env vars as arguments

File: label_studio_ml/server.py [hunk 5]
```diff
@@ -111,13 +131,48 @@ def start_server(args, subprocess_params):
...
+    # configurate project
+    subprocess.check_output(' '.join(["gcloud", "config", "set", "project", project_id]), shell=True)
+    # deploy service
+    subprocess.check_output(' '.join([
+        "gcloud", "run", "deploy",
+        service_name,
+        "--source", output_dir,
+        "--region", region,
+        "--update-env-vars", f"LABEL_STUDIO_ML_BACKEND_V2=1,LABEL_STUDIO_HOSTNAME={args.label_studio_host},LABEL_STUDIO_API_KEY={args.label_studio_api_key}"
+    ]), input=b"y", shell=True)
```
Comment (niklub) [inline label_studio_ml/server.py:160]: could we add some colored message at the end, instructing us to get this links, go on Project Settings -> Machine Learning page and put this link after "Add Model"?

**Review by niklub — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (KonstantinKorotaev) [reply README.md]: Fixed

File: label_studio_ml/examples/simple_text_classifier/requirements.txt (hunk 3 above)
Comment (KonstantinKorotaev) [reply label_studio_ml/examples/simple_text_classifier/requirements.txt]: Fixed

File: label_studio_ml/server.py (hunk 4 above)
Comment (KonstantinKorotaev) [reply label_studio_ml/server.py]: Fixed

**Pull request merged**